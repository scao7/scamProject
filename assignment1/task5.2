(define (f a b c d) (- a b c d))
1000 100 10 1
