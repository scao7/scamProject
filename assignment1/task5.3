(define (g w x y z) (+ w x y z))
25
17
70
-76
