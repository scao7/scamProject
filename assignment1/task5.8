(define (g w x y z) (- w x y z))
-50
-83
10
-69
