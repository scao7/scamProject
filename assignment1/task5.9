(define (g w x y z) (- w x y z))
57
-42
79
-26
