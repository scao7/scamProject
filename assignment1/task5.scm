;task5
(define (curry f )
	(define (lambda a)
     (define (lambda b)
             (define (lambda c)
	                (define (lambda d)
			              (f a b c d)))))
						  )
(define (main)
	(setPort (open (getElement ScamArgs 1) 'read))
	(define arg (eval (readExpr) this))
	(define arg1 (eval (readExpr) this))
	(define arg2 (eval (readExpr) this))
	(define arg3(eval (readExpr) this))
	(define arg4 (eval (readExpr) this))
	;(inspect arg )
	;(inspect arg1)
	;(inspect arg2)
	;(inspect arg3)
	;(inspect arg4)
	;(inspect(((((curry arg)arg1)arg2)arg3)arg4))
	(println "(((((curry " arg ") " arg1 ") " arg2 ") " arg3 ") " arg4 ") is " (((((curry arg)arg1)arg2)arg3)arg4) )
	
  )
