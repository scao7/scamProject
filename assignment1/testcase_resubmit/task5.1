(define (f a b c d) (+ a b c d))
1.1 2.2 3.3 4.4
