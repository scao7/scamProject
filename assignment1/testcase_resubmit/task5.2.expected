(((((curry <function f(a b c d)>) 1000) 100) 10) 1) is 889
